### PR TITLE
Derive Bounded and From<*TableColumn> for usize using EnumIter (strum)

### LINKED ISSUE #17: Derive Ext*TableColumn indexing
(Originally posted by @sshine.)

Right now, each `*TableColumn`'s `.into()` for `usize` hardcodes enumeration, and the corresponding `Ext*TableColumn`'s `.into()` for `usize` does, too, starting with a constant that can go out of sync.

Instead, have two `*Column` that are simple enums (not sum types), derive their `.into()`s, and combine them in a third sum type that piggy-backs on the derived instances. This also makes property tests ("for all columns") possible.

The main change here is that deriving enum instances doesn't work for generic sum types, only for plain enum types (sum types without constructor arguments).

(Originally issue number 38 in the internal issue tracker.)

### PR DESCRIPTION
**Pull request by sshine**
This addresses and closes #17.

Changing the columns (adding, removing, re-ordering) does not cause sporadic indexing errors because some trait instance for some column wasn't updated.

For each table, enums exist that enable addressing a table row by the name of the column rather than its arbitrary number in the column order. This makes the VM and its constraint polynomials more readable and less brittle.

By changing the enums so there are two simple ones, e.g.:
- `BaseProcessorTableColumn` (containing base columns)
- `ExtProcessorTableColumn` (containing ext columns)

and one combined one:

```rust
pub enum ProcessorTableColumn {
    Base(BaseProcessorTableColumn),
    Ext(ExtProcessorTableColumn),
}
```

When `BaseProcessorTableColumn` and `ExtProcessorTableColumn` derive `EnumIter`, it means that `From` and `Bounded` can be implemented in terms of a derived order, rather than one that is manually specified.

This was not possible when `ExtProcessorTableColumn` had a constructor that took an argument, like so:

```rust
pub enum ExtProcessorTableColumn {
    BaseColumn(ProcessorTableColumn),
    ...
}
```

because the `EnumIter` derivation only works for simple enums without arguments. The combined `ProcessorTableColumn`'s trait instances now relies on the derived instances of its inner parameters.

### CODE REVIEW

File: triton-vm/src/table/table_column.rs
Comment (sshine) [inline triton-vm/src/table/table_column.rs:194]: This combined enum seems entirely unnecessary.

We only appear to be using the `*TableColumn` types for
- conversion to `usize` when indexing rows
- conversion via `Display` in the stepping debugger

There appears to be no context where we interchangeably refer to either base columns and extension columns. That could have been the case for constraint polynomials that use the extension columns and the base columns, but in that case, the `usize::from(...)` is sufficient on each enum separately.

**Comment by jan-ferdinand** [comment]
Nice!
Mind pushing this to a feature branch on this repo so I can continue?

**Review by jan-ferdinand — APPROVED** [review]

**Review by sshine — COMMENTED** [review]
Before merging this,
- [x] Get rid of the combined `enum ProcessorTableColumn`
- [x] Derive existing `*TableColumn` enums the same way as processor table.
- [ ] Address all occurrences of `X as usize` vs. `usize::from(X)`.

**Comment by sshine** [comment]
As @jan-ferdinand suggests, we wait until #68 is merged before applying the change to all tables.

**Comment by sshine** [comment]
This PR can be squashed now.

**Pull request merged**
